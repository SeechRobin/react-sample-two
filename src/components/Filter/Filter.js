import React from "react";
import PropTypes from "prop-types";

import DropdownInput from "../UI/DropdownInput/DropdownInput";

const filter = ({ categories, values, handleChange, handleSubmit }) => (
  <div className="filter-wrapper">
    <form>
      <p className="input-category-wrapper">
        <label>Category</label>
        <DropdownInput
          options={categories}
          values={values}
          onChange={handleChange}
        />
      </p>
      <p className="input-location-wrapper">
        <label>Postcode</label>
        <input
          name="location"
          type="text"
          placeholder="Post Code"
          value={values.location}
          onChange={handleChange}
        />
      </p>

      <p className="input-button-wrapper">
        <button className="filter-button" onClick={handleSubmit}>
          Submit
        </button>
      </p>
    </form>
  </div>
);

filter.propTypes = {
  categories: PropTypes.array,
  values: PropTypes.object,
  handleChange: PropTypes.func,
  handleSubmit: PropTypes.func
};

export default filter;
