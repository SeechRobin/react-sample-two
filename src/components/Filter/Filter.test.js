import React from "react";
import { shallow } from "enzyme";

import Filter from "./Filter";

describe("<Filter />", () => {
  let wrapper;
  let handleChangeMock;
  let handleSubmitMock;

  beforeEach(() => {
    const defaultProps = {
      options: [{ id: 1, name: "pro1" }, { id: 2, name: "pro2" }],
      values: { category_id: 0 }
    };

    handleChangeMock = jest.fn();
    handleSubmitMock = jest.fn();
    wrapper = shallow(
      <Filter
        {...defaultProps}
        handleChange={handleChangeMock}
        handleSubmit={handleSubmitMock}
      />
    );
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("clicking submit button", () => {
    const buttonElement = wrapper.find("button");

    buttonElement.simulate("click");
    expect(handleSubmitMock).toBeTruthy();
  });
});
