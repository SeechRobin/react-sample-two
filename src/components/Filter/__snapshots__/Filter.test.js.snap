// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <filter
    handleChange={[MockFunction]}
    handleSubmit={[MockFunction]}
    options={
      Array [
        Object {
          "id": 1,
          "name": "pro1",
        },
        Object {
          "id": 2,
          "name": "pro2",
        },
      ]
    }
    values={
      Object {
        "category_id": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <p
          className="input-category-wrapper"
        >
          <label>
            Category
          </label>
          <DropDownInput
            onChange={[MockFunction]}
            values={
              Object {
                "category_id": 0,
              }
            }
          />
        </p>
        <p
          className="input-location-wrapper"
        >
          <label>
            Postcode
          </label>
          <input
            name="location"
            onChange={[MockFunction]}
            placeholder="Post Code"
            type="text"
          />
        </p>
        <p
          className="input-button-wrapper"
        >
          <button
            className="filter-button"
            onClick={[MockFunction]}
          >
            Submit
          </button>
        </p>
      </form>,
      "className": "filter-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="input-category-wrapper"
          >
            <label>
              Category
            </label>
            <DropDownInput
              onChange={[MockFunction]}
              values={
                Object {
                  "category_id": 0,
                }
              }
            />
          </p>,
          <p
            className="input-location-wrapper"
          >
            <label>
              Postcode
            </label>
            <input
              name="location"
              onChange={[MockFunction]}
              placeholder="Post Code"
              type="text"
            />
          </p>,
          <p
            className="input-button-wrapper"
          >
            <button
              className="filter-button"
              onClick={[MockFunction]}
            >
              Submit
            </button>
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Category
              </label>,
              <DropDownInput
                onChange={[MockFunction]}
                values={
                  Object {
                    "category_id": 0,
                  }
                }
              />,
            ],
            "className": "input-category-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Category",
              },
              "ref": null,
              "rendered": "Category",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onChange": [MockFunction],
                "options": undefined,
                "values": Object {
                  "category_id": 0,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Postcode
              </label>,
              <input
                name="location"
                onChange={[MockFunction]}
                placeholder="Post Code"
                type="text"
              />,
            ],
            "className": "input-location-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Postcode",
              },
              "ref": null,
              "rendered": "Postcode",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "location",
                "onChange": [MockFunction],
                "placeholder": "Post Code",
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="filter-button"
              onClick={[MockFunction]}
            >
              Submit
            </button>,
            "className": "input-button-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "filter-button",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
          "type": "p",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <p
            className="input-category-wrapper"
          >
            <label>
              Category
            </label>
            <DropDownInput
              onChange={[MockFunction]}
              values={
                Object {
                  "category_id": 0,
                }
              }
            />
          </p>
          <p
            className="input-location-wrapper"
          >
            <label>
              Postcode
            </label>
            <input
              name="location"
              onChange={[MockFunction]}
              placeholder="Post Code"
              type="text"
            />
          </p>
          <p
            className="input-button-wrapper"
          >
            <button
              className="filter-button"
              onClick={[MockFunction]}
            >
              Submit
            </button>
          </p>
        </form>,
        "className": "filter-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="input-category-wrapper"
            >
              <label>
                Category
              </label>
              <DropDownInput
                onChange={[MockFunction]}
                values={
                  Object {
                    "category_id": 0,
                  }
                }
              />
            </p>,
            <p
              className="input-location-wrapper"
            >
              <label>
                Postcode
              </label>
              <input
                name="location"
                onChange={[MockFunction]}
                placeholder="Post Code"
                type="text"
              />
            </p>,
            <p
              className="input-button-wrapper"
            >
              <button
                className="filter-button"
                onClick={[MockFunction]}
              >
                Submit
              </button>
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  Category
                </label>,
                <DropDownInput
                  onChange={[MockFunction]}
                  values={
                    Object {
                      "category_id": 0,
                    }
                  }
                />,
              ],
              "className": "input-category-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Category",
                },
                "ref": null,
                "rendered": "Category",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onChange": [MockFunction],
                  "options": undefined,
                  "values": Object {
                    "category_id": 0,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  Postcode
                </label>,
                <input
                  name="location"
                  onChange={[MockFunction]}
                  placeholder="Post Code"
                  type="text"
                />,
              ],
              "className": "input-location-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Postcode",
                },
                "ref": null,
                "rendered": "Postcode",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "location",
                  "onChange": [MockFunction],
                  "placeholder": "Post Code",
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="filter-button"
                onClick={[MockFunction]}
              >
                Submit
              </button>,
              "className": "input-button-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "filter-button",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
            "type": "p",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
